src/ucp/core/ucp_request.c

```diff
@@ -181,24 +181,25 @@ int ucp_request_pending_add(ucp_request_t *req, ucs_status_t *req_status)
 }
 
 static void ucp_request_dt_dereg(ucp_context_t *context, ucp_dt_reg_t *dt_reg,
-                                 size_t count, ucp_request_t *req_dbg)
+                                 size_t count, uct_memory_type_t mem_type,
+                                 ucp_request_t *req_dbg)
 {
     size_t i;
 
     for (i = 0; i < count; ++i) {
         ucp_trace_req(req_dbg, "mem dereg buffer %ld/%ld md_map 0x%"PRIx64,
                       i, count, dt_reg[i].md_map);
-        ucp_mem_rereg_mds(context, 0, NULL, 0, 0, NULL, UCT_MD_MEM_TYPE_HOST, NULL,
+        ucp_mem_rereg_mds(context, 0, NULL, 0, 0, NULL, mem_type, NULL,
```

Comment:
maybe we can just pass UCT_MD_MEM_TYPE_HOST here as before? the mem type is not really required by ucp_mem_rereg_mds if it's not registering anything (can update the doc of ucp_mem_rereg_mds to indicate that)